types.go

```diff
@@ -55,6 +55,11 @@ type (
 		// Checksum ensures the integrity of packages
 		// refereced by URL. Ignored for literals.
 		Checksum Checksum `json:"checksum"`
+	}
+
+	PackageRef struct {
+		PackageName      string
+		PackageNamespace string
```

Comment:
Versioning belongs in a higher-level resource, IMO: we can create a `VersionedFunction` or a `VersionedPackage` and have a controller loop that watches git and creates `Functions` or `Packages`.

The entrypoint is what allows different functions to share a package. So it belongs either in FunctionSpec or PackageRef. Perhaps we should call it something else? "Handler" maybe?